oneview/resource_logical_enclosure.go

```diff
@@ -0,0 +1,358 @@
+					Schema: map[string]*schema.Schema{
+						"firmware_baseline_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"firmware_update_on": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"force_install_firmware": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"logical_interconnect_update_mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"update_firmware_on_unmanaged_interconnect": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"validate_if_li_firmware_update_is_non_disruptive": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"ip_addressing_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"ipv4_range": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dns_servers": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"domain": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"gateway": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"ip_range_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"subnet_mask": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"logical_interconnect_uris": {
+				Optional: true,
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"power_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"scaling_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"scopes_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+
+}
+func resourceLogicalEnclosureCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	logicalEnclosure := ov.LogicalEnclosure{
+		Name:                   d.Get("name").(string),
+		EnclosureGroupUri:      utils.NewNstring(d.Get("enclosure_group_uri").(string)),
+		AmbientTemperatureMode: d.Get("ambient_temperature_mode").(string),
+		DeleteFailed:           d.Get("delete_failed").(bool),
+		ScopesUri:              utils.NewNstring(d.Get("scopes_uri").(string)),
+		Type:                   d.Get("type").(string),
+		Status:                 d.Get("status").(string),
+		URI:                    utils.NewNstring(d.Get("uri").(string)),
```

Comment:
The attributes for Create are mismatching when compared with documentation. Can you check once?

Reply:
Addressed the comment, changed the structure as per API.